### PR TITLE
Added disabled spoken forms error

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: packages/common/src/types/TalonSpokenForms.ts [hunk 1]
```diff
@@ -42,3 +42,10 @@ export class NeedsInitialTalonUpdateError extends Error {
     this.name = "NeedsInitialTalonUpdateError";
   }
 }
+
+export class DisabledSpokenFormsError extends Error {
```
Comment (pokey) [inline packages/common/src/types/TalonSpokenForms.ts:46]:
```suggestion
export class `DisabledCustomSpokenFormsError` extends Error {
```

File: packages/common/src/types/TalonSpokenForms.ts [hunk 2]
```diff
@@ -42,3 +42,10 @@ export class NeedsInitialTalonUpdateError extends Error {
...
+  constructor() {
+    super("Spoken forms are disabled");
```
Comment (pokey) [inline packages/common/src/types/TalonSpokenForms.ts:48]:
```suggestion
    super("Custom spoken forms are not currently supported in this ide");
```

File: packages/cursorless-engine/src/spokenForms/CustomSpokenForms.ts [hunk 3]
```diff
@@ -81,6 +82,9 @@ export class CustomSpokenForms {
       if (err instanceof NeedsInitialTalonUpdateError) {
         // Handle case where spokenForms.json doesn't exist yet
         this.needsInitialTalonUpdate_ = true;
+      }
+      if (err instanceof DisabledSpokenFormsError) {
+        // Handle case where spoken forms are disabled
```
Comment (pokey) [inline packages/cursorless-engine/src/spokenForms/CustomSpokenForms.ts]:
```suggestion
        // Do nothing: this ide doesn't currently support custom spoken forms
```

**Comment by AndreasArvidsson** [comment]
@pokey Suggestions accepted

File: packages/cursorless-engine/src/spokenForms/CustomSpokenForms.ts [hunk 4]
```diff
@@ -81,6 +82,9 @@ export class CustomSpokenForms {
       if (err instanceof NeedsInitialTalonUpdateError) {
         // Handle case where spokenForms.json doesn't exist yet
         this.needsInitialTalonUpdate_ = true;
+      }
+      if (err instanceof DisabledCustomSpokenFormsError) {
```
Comment (pokey) [inline packages/cursorless-engine/src/spokenForms/CustomSpokenForms.ts:86]:
```suggestion
      } else if (err instanceof DisabledCustomSpokenFormsError) {
```

File: packages/common/src/types/TalonSpokenForms.ts [hunk 5]
```diff
@@ -42,3 +42,10 @@ export class NeedsInitialTalonUpdateError extends Error {
     this.name = "NeedsInitialTalonUpdateError";
   }
 }
+
+export class DisabledCustomSpokenFormsError extends Error {
+  constructor() {
+    super("Custom spoken forms are not currently supported in this ide");
+    this.name = "DisabledSpokenFormsError";
```
Comment (pokey) [inline packages/common/src/types/TalonSpokenForms.ts:49]:
```suggestion
    this.name = "DisabledCustomSpokenFormsError";
```

**Review by pokey — APPROVED** [review]

**Pull request merged**